File: lib/core/steam/steam_account.dart
/// A single connected Steam account (own Web-API key + the SteamID64 it was
/// used to sign into). The library can merge games from several of these so
/// a user with more than one Steam account sees one combined library.
class SteamAccount {
  final String steamId;
  final String apiKey;
  final String personaName;
  final String avatarUrl;

  SteamAccount({
    required this.steamId,
    required this.apiKey,
    required this.personaName,
    required this.avatarUrl,
  });

  factory SteamAccount.fromJson(Map<String, dynamic> json) => SteamAccount(
    steamId: json['steamId'] as String,
    apiKey: json['apiKey'] as String,
    personaName: (json['personaName'] as String?) ?? 'Steam-Nutzer',
    avatarUrl: (json['avatarUrl'] as String?) ?? '',
  );

  Map<String, dynamic> toJson() => {
    'steamId': steamId,
    'apiKey': apiKey,
    'personaName': personaName,
    'avatarUrl': avatarUrl,
  };
}

File: ios-native/GameLib/Models/SteamAccount.swift
import Foundation

/// A single connected Steam account (own Web API key + the SteamID64 it was
/// used to sign into). Mirrors the Flutter app's SteamAccount model so the
/// same shape round-trips through Cloud-Sync/QR transfer once that phase
/// lands — field names intentionally match the Dart `toJson()`/`fromJson()`.
struct SteamAccount: Codable, Identifiable, Equatable, Sendable {
    var steamId: String
    var apiKey: String
    var personaName: String
    var avatarUrl: String

    var id: String { steamId }
}
